Parted uses GNU89 inline semantics, but that breaks with GCC 4.3 but the
option to GCC wasn't intruduced until GCC 4.1.3. So we add a check to
autotools to dynamically enable it, if it is supported.

Signed-off-by: Jonas Karlsson
--- parted-1.8.6/configure.ac	2007-03-20 17:24:48.000000000 +0000
+++ parted-1.8.6/configure.ac.new	2008-05-13 01:57:57.000000000 +0000
@@ -196,6 +196,17 @@
 libraries.)
 	AM_DISABLE_SHARED
 fi
+
+AC_MSG_CHECKING([whether gcc supports -fgnu89-inline])
+	_gcc_cflags_save="$CFLAGS"
+CFLAGS="${CFLAGS} -fgnu89-inline"
+AC_COMPILE_IFELSE(AC_LANG_PROGRAM([]),
+	_gcc_gnu89_inline=yes, _gcc_gnu89_inline=no)
+AC_MSG_RESULT($_gcc_gnu89_inline)
+AM_CONDITIONAL([HAVE_GCC_GNU89_INLINE_OPTION],
+	[test "x$_gcc_gnu89_inline" = "xyes"])
+CFLAGS=$_gcc_cflags_save;
+
 AM_PROG_LIBTOOL
 
 AM_GNU_GETTEXT_VERSION([0.15])
--- parted-1.8.6/Makefile.am	2007-03-16 20:09:52.000000000 +0000
+++ parted-1.8.6/Makefile.am.new	2008-05-13 01:59:52.000000000 +0000
@@ -18,6 +18,13 @@
 pcdir = $(libdir)/pkgconfig
 pc_DATA = libparted.pc
 
+# The -fgnu89-inline' option appeared in GCC 4.1.3.
+if HAVE_GCC_GNU89_INLINE_OPTION
+# Parted currently rely on GNU inline semantics, not C99 inline.
+	AM_CFLAGS += -fgnu89-inline
+  
+endif HAVE_GCC_GNU89_INLINE_OPTION
+  
 # This is best not done via configure.ac, because automake's
 # make distcheck target does not like auto-generated files
 # being included in the distributed archive.
